### PR TITLE
"all" type search query spec

### PR DESCRIPTION
**Pull request by navis**
This is really a trivial patch for easily accepting all values from search query. we can use {"type: "all"} instead of {"type": "contains", "value": ""} or something.

### CODE REVIEW

**Comment by gianm** [comment]
transient failure, bouncing.

File: processing/src/main/java/io/druid/query/search/search/SearchQuerySpec.java [hunk 1]
```diff
@@ -29,11 +29,27 @@
...
+    @JsonSubTypes.Type(name = "regex", value = RegexSearchQuerySpec.class),
+    @JsonSubTypes.Type(name = "all", value = SearchQuerySpec.TakeAll.class)
 })
 public interface SearchQuerySpec
 {
   public boolean accept(String dimVal);
 
   public byte[] getCacheKey();
+
+  class TakeAll implements SearchQuerySpec
```
Comment (gianm) [inline processing/src/main/java/io/druid/query/search/search/SearchQuerySpec.java]: Even though this is simple, I'd prefer it being called `AllSearchQuerySpec` and being in its own file for consistency. please also define hashCode, equals, toString.

File: processing/src/main/java/io/druid/query/search/search/SearchQuerySpec.java [hunk 2]
```diff
@@ -29,11 +29,27 @@
...
+    @Override
+    public boolean accept(String dimVal)
+    {
+      return true;
+    }
+
+    @Override
+    public byte[] getCacheKey()
+    {
+      return new byte[]{0x7f};
```
Comment (fjy) [inline processing/src/main/java/io/druid/query/search/search/SearchQuerySpec.java]: can we please define this as a constant?

File: processing/src/main/java/io/druid/query/search/search/SearchQuerySpec.java [hunk 3]
```diff
@@ -29,11 +29,27 @@
     @JsonSubTypes.Type(name = "contains", value = ContainsSearchQuerySpec.class),
     @JsonSubTypes.Type(name = "insensitive_contains", value = InsensitiveContainsSearchQuerySpec.class),
     @JsonSubTypes.Type(name = "fragment", value = FragmentSearchQuerySpec.class),
-    @JsonSubTypes.Type(name = "regex", value = RegexSearchQuerySpec.class)
+    @JsonSubTypes.Type(name = "regex", value = RegexSearchQuerySpec.class),
+    @JsonSubTypes.Type(name = "all", value = SearchQuerySpec.TakeAll.class)
```
Comment (fjy) [inline processing/src/main/java/io/druid/query/search/search/SearchQuerySpec.java]: just for cleaniness I think it'd be nice to have this in a separate file

**Comment by gianm** [comment]
looks good, with one comment.

**Comment by navis** [comment]
@gianm addressed comments. thanks.

**Comment by fjy** [comment]
👍

**Comment by gianm** [comment]
👍 , LGTM, unit test would be good too though.

**Comment by navis** [comment]
added UT

**Pull request merged**